blobstore/src/main/java/org/jclouds/blobstore/domain/StorageMetadata.java

```diff
@@ -92,4 +92,7 @@
     */
    Date getLastModified();
 
+   // TODO: how does this interact with ContentMetadata.getContentLength?
```

Comment:
This is the core issue with this pull request: where should size exist?  All providers return it as part of the object metadata, but traditionally jclouds has stored it in the content metadata.  Listing operations return the former and per-object operations return the latter.  This really should only exist in the former; should we deprecate the latter?